integration/sql/impl/src/visitor.rs

```diff
@@ -699,3 +754,60 @@ fn get_table_name_from_table_factor(table: &TableFactor) -> Result<Vec<Ident>> {
         ))
     }
 }
+
+pub fn extract_up_to_two_ident_values(
+    object_name: &ObjectName,
+) -> (Option<String>, Option<String>) {
+    let mut idents_iter = object_name.0.iter();
+
+    let first_value = idents_iter.next().map(|ident| ident.value.clone());
+    let second_value = idents_iter.next().map(|ident| ident.value.clone());
+
+    (first_value, second_value)
+}
+
+#[test]
+fn test_extract_up_to_two_ident_values_one_ident() {
+    let object_name = ObjectName(vec![Ident::new("first")]);
+    assert_eq!(
+        extract_up_to_two_ident_values(&object_name),
+        (Some("first".to_string()), None)
+    );
+}
+
+#[test]
+fn test_extract_up_to_two_ident_values_one_ident_with_quotes() {
+    let object_name = ObjectName(vec![Ident::with_quote('`', "first")]);
+    assert_eq!(
+        extract_up_to_two_ident_values(&object_name),
+        (Some("first".to_string()), None)
+    );
+}
+
+#[test]
+fn test_extract_up_to_two_ident_values_two_idents() {
+    let object_name = ObjectName(vec![Ident::new("first"), Ident::with_quote('`', "second")]);
+    assert_eq!(
+        extract_up_to_two_ident_values(&object_name),
+        (Some("first".to_string()), Some("second".to_string()))
+    );
+}
+
+#[test]
+fn test_extract_up_to_two_ident_values_three_idents() {
+    let object_name = ObjectName(vec![
+        Ident::new("first"),
+        Ident::with_quote('`', "second"),
+        Ident::new("third"),
+    ]);
+    assert_eq!(
+        extract_up_to_two_ident_values(&object_name),
+        (Some("first".to_string()), Some("second".to_string()))
+    );
+}
+
+#[test]
+fn test_extract_up_to_two_ident_values_empty_idents() {
+    let object_name = ObjectName(vec![]);
```

Comment:
Also, not sure where to put these tests of a helper function

Reply:
I think helper function tests are fine in module. That's what Rust recommends